docs/apidiffs/current_vs_latest/opentelemetry-sdk.txt

```diff
@@ -4,6 +4,7 @@ Comparing source compatibility of  against
 	+++  NEW METHOD: PUBLIC(+) io.opentelemetry.api.metrics.MeterProvider getMeterProvider()
 	+++  NEW METHOD: PUBLIC(+) io.opentelemetry.sdk.logs.SdkLogEmitterProvider getSdkLogEmitterProvider()
 	+++  NEW METHOD: PUBLIC(+) io.opentelemetry.sdk.metrics.SdkMeterProvider getSdkMeterProvider()
+	+++  NEW METHOD: PUBLIC(+) java.lang.String toString()
```

Comment:
this is weird that jamicmp flags this, since `toString()` is on `Object`. 🤔

Reply:
I believe it's because now it is possible for code to compile against `SdkMeterProvider.toString()` directly, so after adding these we can't remove them.

Reply:
oh interesting i hadn't thought about that before 👍

Reply:
I'm missing something. Can you give me an example of what you mean?

Reply:
If my app wrote this,
```
SdkMeterProvider provider = SdkMeterProvider.builder().build();
System.out.println(provider.toString());
```

Then the bytecode would link to `SdkMeterProvider.toString` since the concrete type of `provider` is `SdkMeterProvider`. This wouldn't be true for `System.out.println(provider)`, where within println the concrete type is `Object` and it links to `Object.toString()`, which virtually gets resolved to the subclass. A Java strangeness indeed

Reply:
won't the runtime walk up the hierarchy either way, though? If this is true, I'm extremely surprised.

Reply:
If you mean it would walk up to `Object.toString` if the bytecode said `invokevirtual SdkMeterProvider.toString` but we removed that method, no I don't think so, it's looking for that method only and will throw a link error.

Reply:
I just tested this locally and it works as I expected. The runtime will walk up the hierarchy and call the parent implementation.

Reply:
@jkwatson Showed me the bytecode too - sorry for red herring, was completely wrong and surprised by this!

Reply:
thx @jkwatson 🧐